kimchi/src/circuits/gate.rs

```diff
@@ -87,21 +108,26 @@ impl<F: Field> SingleLookup<F> {
 /// A spec for checking that the given vector belongs to a vector-valued lookup table.
 #[derive(Clone, Serialize, Deserialize)]
 pub struct JointLookup<F> {
-    pub table_id: usize,
+    pub table_id: i32,
```

Comment:
They're not strictly necessary, but for ease of use I was planning to have negative IDs for 'runtime' tables and positive ones for the 'fixed' tables. We currently hard-code the table ID for XOR, this was mostly a workaround so that we don't have to account for that / any other fixed tables that we add.

I'll add a comment to this effect.